src/aiodynamo/errors.py

```diff
@@ -188,6 +197,17 @@ def exception_from_response(status: int, body: bytes) -> Exception:
         return ServiceUnavailable()
     try:
         data = json.loads(body)
-        return ERRORS[data["__type"].split("#", 1)[1]](data)
+        error = ERRORS[data["__type"].split("#", 1)[1]](data)
+        if isinstance(error, TransactionCanceled):
+            error = extract_error_from_transaction_canceled(data)
+        return error
     except Exception:
         return UnknownError(status, body)
+
+
+def extract_error_from_transaction_canceled(data: Dict[str, Any]) -> AIODynamoError:
+    try:
+        error = data["CancellationReasons"][0]
+        return ERRORS[f"{error['Code']}Exception"](error["Message"])
+    except Exception:
+        return ERRORS[data["__type"].split("#", 1)[1]](data)
```

Comment:
`IdempotentParameterMismatchException` and `TransactionInProgressException` are not yet in `ERRORS` but can happen according to the docs

Reply:
Nice catch! Just added these exceptions 😃